accession-commons-jpa/src/test/java/uk/ac/ebi/ampt2d/commons/accession/persistence/jpa/services/JpaBasicSpringDataRepositoryDatabaseServiceTest.java

```diff
@@ -376,11 +384,37 @@ public void testMergeAccessionDeprecatedDestination() throws AccessionDoesNotExi
             AccessionDeprecatedException, AccessionMergedException {
         service.save(Arrays.asList(new AccessionWrapper("a1", "h1", TestModel.of("something1"), 1)));
         service.save(Arrays.asList(new AccessionWrapper("a2", "h2", TestModel.of("something2"), 1)));
-        assertEquals(1, service.findAllByAccession(Arrays.asList("a1")).size());
-        assertEquals(1, service.findAllByAccession(Arrays.asList("a2")).size());
+        assertTrue(findLastVersionByAccession("a1").isPresent());
+        assertTrue(findLastVersionByAccession("a2").isPresent());
         service.deprecate("a2", "blah");
 
         service.merge("a1", "a2", "reasons");
     }
 
+    @Test
+    public void testMultipleMergeAccession() throws AccessionDoesNotExistException,
+            AccessionDeprecatedException, AccessionMergedException {
+        service.save(Arrays.asList(new AccessionWrapper("a1", "h1", TestModel.of("something1"), 1)));
+        service.save(Arrays.asList(new AccessionWrapper("a2", "h2", TestModel.of("something2"), 1)));
+        service.save(Arrays.asList(new AccessionWrapper("a3", "h3", TestModel.of("something3"), 1)));
+        assertTrue(findLastVersionByAccession("a1").isPresent());
+        assertTrue(findLastVersionByAccession("a2").isPresent());
+        assertTrue(findLastVersionByAccession("a3").isPresent());
+        service.merge("a1", "a2", "reasons");
+        service.merge("a2", "a3", "reasons");
+        assertEquals("a3",findLastVersionByAccession("a1").get().getAccession());
+        assertEquals("a3",findLastVersionByAccession("a2").get().getAccession());
+        assertEquals("a3",findLastVersionByAccession("a3").get().getAccession());
+    }
+
+    private Optional<AccessionWrapper<TestModel, String, String>> findLastVersionByAccession(String accession) {
```

Comment:
I don't think this method adds more clarity, as you can test presence with `assertNotNull`. It also prevents the actual API from being tested because you add a wrapper on top of it, and even hides any exception thrown by just printing the stack trace.

Reply:
+1